SUBLINKS: hide sublinks by default and toggle showing them

ajwl opened the pull request:
## What's changed?
Previously all sublinks on a story were shown. This change hides them unless the user wants to see them. 

If the user drags in a story over a closed set of sublinks: they should highlight a darker grey then open after a 0.75 second pause and then allow for the story to be drop in there

**New view**
(closed)
<img width="592" alt="Screenshot 2019-04-26 at 10 25 28" src="https://user-images.githubusercontent.com/10324129/56797928-ba3e0100-680d-11e9-98e8-ce12dca09a5f.png">


(open)
<img width="592" alt="Screenshot 2019-04-26 at 10 25 51" src="https://user-images.githubusercontent.com/10324129/56797947-c5912c80-680d-11e9-87bc-c04756fbffae.png">

**On Clipboard**
<img width="164" alt="Screenshot 2019-04-26 at 16 27 08" src="https://user-images.githubusercontent.com/10324129/56818802-4e28c080-6840-11e9-8572-bb13ed5929f6.png">

<img width="164" alt="Screenshot 2019-04-26 at 16 27 16" src="https://user-images.githubusercontent.com/10324129/56818816-5aad1900-6840-11e9-9c3c-11da493f3b99.png">



Trello ticket: https://trello.com/c/UGcKTkyE/516-collapse-sublinks-in-the-stories


## Implementation notes
Needed to change integration test!
Comment explains - depending on whether the article dragged in is a sublink or a full article, the number of new dropzones created is different. 
- Only one new dropzone is add when a sublink is pulled in
- Two are created when a full article is pulled in.

## Checklist

### General
- [ ] 🤖 Relevant tests added
- [x] ✅ CI checks / tests run locally
- [x] 🔍 Checked on CODE

### Client
- [x] 🚫 No obvious console errors on the client (i.e. React dev mode errors)
- [x] 🎛️ No regressions with existing user interactions (i.e. all existing buttons, inputs etc. work)
- [x] 📷 Screenshots / GIFs of relevant UI changes included

ajwl commented:
@RichieAHB @jonathonherbert couple of quick questions:

1) Should i be setting up the style etc for the Sublink link in this component? I feel like it should be down a level or two, but am not sure where exactly. 

2) CollectionItemMetaContainer requires children. But i want it to be empty? How can I pass it an empty children attribute?

jonathonherbert commented:
Re: 2., we can make `children` optional by adding a question mark -- `{ children }: { children?: React.ReactNode }`

Re: 1., I'm not quite sure what you mean but perhaps we could chat tomorrow?

ajwl commented:
1) Added drag behaviour with a pause and a grey highlight as requested by Akemi
2) Changed an integration test to work for new behaviour 
@RichieAHB @Reettaphant 
Live on code -> if you want to check

akemitakagi commented:
@ajwl I am sorry I didn't spot this earlier. The arrow icons are wrong way around. When closed the icon should point down. When opened the icon should point up. Zeplin link: zpl.io/bzN0WJM

ajwl commented:
@akemitakagi ah sure - easy to change

ajwl commented on client-v2/src/components/FrontsEdit/CollectionComponents/CollectionItem.tsx:
```diff
@@ -112,7 +132,14 @@ class CollectionItem extends React.Component<ArticleContainerProps> {
+            />
+            {numSupportingArticles === 0
+              ? children
+              : this.state.showArticleSublinks && children}
```
@RichieAHB this could probably be clarified..

ajwl commented:
Good point Rich - that doesn't look quite right on clipboard. I'll do some style changes

ajwl commented on client-v2/src/components/FrontsEdit/CollectionComponents/CollectionItem.tsx:
```diff
@@ -132,6 +162,14 @@ const createMapStateToProps = () => {
+    );
+    let numSupportingArticles = 0;
+    if (maybeArticle && maybeArticle.meta && maybeArticle.meta.supporting) {
+      numSupportingArticles = maybeArticle.meta.supporting.length;
```
Is this what you want to change? @RichieAHB

ajwl commented on client-v2/src/components/Clipboard.tsx:
```diff
@@ -83,13 +83,20 @@ const StyledDragIntentContainer = styled(DragIntentContainer)`
+  margin: 8px -10px 4px;
+  width: 115%;
+`;
+
```
Design wanted a general clipboard change so there is a full-width grey line under each story in the clipboard. This `115%` stuff is hacky I know but works without changing the padding everywhere else.

ajwl commented on client-v2/src/components/Clipboard.tsx:
```diff
@@ -192,57 +199,64 @@ class Clipboard extends React.Component<ClipboardProps> {
+                          </ArticleFragmentLevel>
+                        </CollectionItem>
+                      </ArticleWrapper>
+                      <FullDivider />
```
The addition of `<FullDivider />` is the only change here, the rest is just indentation..

ajwl commented:
Have just noticed one final bug --> opening the sublinks in Clipboard triggers an open of the edit panel, which is not desirable behviour. Guess the solution is to make a click zone smaller

RichieAHB commented on client-v2/src/components/FrontsEdit/CollectionComponents/CollectionItem.tsx:
```diff
@@ -97,23 +114,41 @@ class CollectionItem extends React.Component<ArticleContainerProps> {
+              showArticleSublinks={this.state.showArticleSublinks}
+              parentId={parentId}
+            />
+            {/* If there are no supporting articles, the children still need to be rendered, because the dropzone is a child  */}
```
👍
